codegen/smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/PaginatorGeneratorTest.kt

```diff
@@ -263,6 +263,130 @@ class PaginatorGeneratorTest {
+            operation ListFlows {
+                input: ListFlowsRequest,
+                output: ListFlowsResponse
+            }
+            
+            structure ListFlowsRequest {
+                @httpQuery("FlowVersion")
+                FlowVersion: String,
+                @httpQuery("Marker")
+                Marker: String,
+                @httpQuery("MasterRegion")
+                MasterRegion: String,
+                @httpQuery("MaxItems")
+                MaxItems: Integer
+            }
+            
+            structure ListFlowsResponse {
+                Flows: FlowList,
+                NextMarker: String
+            }
+            
+            list FlowList {
+                member: Flow
+            }
+            
+            structure Flow {
+                Name: String
+            }
+        """.toSmithyModel()
+        val testContextWithItems = testModelWithItems.newTestContext("FlowService", "com.test")
+
+        val codegenContextWithItems = object : CodegenContext {
+            override val model: Model = testContextWithItems.generationCtx.model
+            override val symbolProvider: SymbolProvider = testContextWithItems.generationCtx.symbolProvider
+            override val settings: KotlinSettings = testContextWithItems.generationCtx.settings
+            override val protocolGenerator: ProtocolGenerator = testContextWithItems.generator
+            override val integrations: List<KotlinIntegration> = testContextWithItems.generationCtx.integrations
+        }
+
+        val unit = PaginatorGenerator()
+        unit.writeAdditionalFiles(codegenContextWithItems, testContextWithItems.generationCtx.delegator)
+
+        testContextWithItems.generationCtx.delegator.flushWriters()
+        val testManifest = testContextWithItems.generationCtx.delegator.fileManifest as MockManifest
+        val actual = testManifest.expectFileString("src/main/kotlin/com/test/paginators/Paginators.kt")
+
+        val expectedCode = """
+            /**
+             * Paginate over [ListFlowsResponse] results.
+             *
+             * When this operation is called, a [kotlinx.coroutines.Flow] is created. Flows are lazy (cold) so no service
+             * calls are made until the flow is collected. This also means there is no guarantee that the request is valid
+             * until then. Once you start collecting the flow, the SDK will lazily load response pages by making service
+             * calls until there are no pages left or the flow is cancelled. If there are errors in your request, you will
+             * see the failures only after you start collection.
+             * @param initialRequest A [ListFlowsRequest] to start pagination
+             * @return A [kotlinx.coroutines.flow.Flow] that can collect [ListFlowsResponse]
+             */
+            public fun TestClient.listFlowsPaginated(initialRequest: ListFlowsRequest = ListFlowsRequest { }): kotlinx.coroutines.flow.Flow<ListFlowsResponse> =
```

Comment:
question: These will only be fully qualified `kotlinx.coroutines.flow.Flow` when there's an import collision, right?

Reply:
Yes, although typically `kotlinx.coroutines.flow.Flow` will be referenced _first_ in our codegen and so a conflicting symbol like `com.foo.Flow` would actually be the one which gets the fully-qualified name in code. This test case shows it the other way around because the very first paginator is the one which has the collision and the model shape is referenced before `kotlinx.coroutines.flow.Flow`.